content/posts/2019-08-03---Going-from-ember-to-vue.md

```diff
@@ -162,10 +175,16 @@ This means that child component does not need to receive an action as a paramete
 All it does is emit something, pass any properties along with it and then it stops caring.
 It's up to the parent component to listen to it and respond.
 
-One of the things I disliked with the actions helper in Ember was when the child component could call `this.passedAction()` and I would get an error that it wasn't a function because I forgot to pass it down, the child component stopped working.
-With emitting, the child component can continue to function.
-To be clear: making API calls to your backend should happen through the Vuex store -- mutating applicaton state should not be done with `emit`.
-However, if you have buttons component that toggles something on and off, this is a good case for using `emit` as the parent component is most likely the only thing that cares about the event.
+One of the things I disliked with the actions helper in Ember was that a child component might not always want to bubble something up.
```

Comment:
because of your experience with this, and many others' similar experience, this is discouraged now-a-days

Reply:
By design components are isolated and actions don't bubble up, so I am confused by this.

Reply:
P.S. This point is in strict opposition to the previous one, where emitting always bubbles up the tree. Do I understand that correctly?